src/cmdstan-guide/external_code.Rmd

```diff
@@ -42,20 +47,30 @@ this program to C++, but the generated C++ code will not compile
 unless you write a file
 such as `examples/bernoulli/make_odds.hpp` with the following lines
 ```C++
-#include <stan/model/model_header.hpp>
 #include <ostream>
 
-namespace bernoulli_model_namespace {
-template <typename T0__,
-          stan::require_all_t<stan::is_stan_scalar<T0__>>* = nullptr>
-  stan::promote_args_t<T0__>
-  make_odds(const T0__& theta, std::ostream* pstream__) {
+double
+make_odds(const double &theta, std::ostream *pstream__)
+{
+    return theta / (1 - theta);
+}
+```
+
+The signature for this function needs to fulfill all the usages in the C++
+class emitted by stanc. The `pstream__` argument is mandatory
+in the signature but need not be used if your function does not print
+any output. Because `make_odds` was declared with a `data` argument and only
+used in `generated quantites`, a signature which accepts and returns `double` is
```

Comment:
It might be good to clarify that autodiff is needed if the function is going to take a `parameter` as an input